src/screens/createPost/index.js

```diff
@@ -0,0 +1,204 @@
+      </Button>
+      <DropzoneDialog
+        open={open}
+        onSave={handleSave}
+        acceptedFiles={["image/jpeg", "image/png", "image/bmp"]}
+        showPreviews={true}
+        maxFileSize={5000000}
+        onClose={handleClose}
+      />
+    </div>
+  );
+}
+
+export default function Maxmised(props) {
+  const inputEl1 = useRef(null);
+  const inputEl2 = useRef(null);
+  const classes = useStyles();
+  var imgarr = [];
+  const [text1, setText1] = useState(false);
+  const [text2, setText2] = useState(false);
+
+  return (
+    <Card className={classes.card}>
+      <CardHeader
+        classes={{ title: classes.title, subheader: classes.subheader }}
+        avatar={
+          <Avatar className={classes.avatar}>
+            <CreateIcon color="primary" fontSize="medium" />
+          </Avatar>
+        }
+        title="CREATE POST"
+        subheader={props.stream}
+        style={{
+          textAlign: "left",
+          backgroundColor: "#3f51b5",
+        }}
+      />
+
+      <CardContent>
+        <Typography variant="body2" color="black" component="p">
+          <form className={classes.root} noValidate autoComplete="off">
+            <div>
+              <TextField
+                required
+                id="post-title"
+                label="Post Title"
+                placeholder="Your Title"
+                variant="outlined"
+                fullWidth
+                error={text1}
+                inputRef={inputEl1}
+              />
+            </div>
+            <div>
+              <TextField
+                required
+                id="post-content"
+                label="Post Content"
+                placeholder="Your Content"
+                multiline
+                rows="12"
+                variant="outlined"
+                error={text2}
+                inputRef={inputEl2}
+              />
+            </div>
+            <div className={classes.root}>
+              <DropzoneDialogExample filearray={imgarr} />
+            </div>
+            <div>
+              <Button
+                variant="contained"
+                color="primary"
+                className={classes.button}
+                endIcon={<SendIcon />}
+                onClick={() => {
+                  if (inputEl1.current.value === "") setText1(true);
+                  else setText1(false);
+                  if (inputEl2.current.value === "") setText2(true);
+                  else setText2(false);
+                  if (
+                    !(
+                      inputEl1.current.value === "" ||
+                      inputEl2.current.value === ""
+                    )
+                  ) {
+                    setText1(false);
+                    setText2(false);
+                    props.onSubmit(
+                      inputEl1.current.value,
+                      inputEl2.current.value,
+                      imgarr
+                    );
+                  }
+                }}
```

Comment:
Extract this into a function (inside this component itself), and use that function here. The general code pattern is to keep component logic in the functions and UI in the return value.